packages/sst/src/constructs/NextjsSite.ts

```diff
@@ -245,179 +302,200 @@ export class NextjsSite extends SsrSite {
     );
   }
 
-  protected plan(bucket: Bucket) {
-    const {
-      path: sitePath,
-      edge,
-      experimental,
-      imageOptimization,
-      cdk,
-    } = this.props;
-    const stack = Stack.of(this);
-    const serverConfig = {
-      description: "Next.js server",
-      bundle: path.join(sitePath, ".open-next", "server-function"),
-      handler: "index.handler",
+  public get regionalServersCdk() {
+    if (this.doNotDeploy) return;
+    const regionalServers = this.serverFunctions.reduce((acc, server) => {
+      return {...acc, [
+        server.function ?
+        server.id.replace("ServerFunction", "") :
+        server.id.replace("ServerContainer", "")
+      ] : server.function ?? server.cdk};
+    }, {} as InterpolatedCdkOutputs<ONConfig>)
+    return regionalServers
+  }
+
+  private createFunctionOrigin(fn: OpenNextFunctionOrigin, key: string, bucket: Bucket) : FunctionOriginConfig {
+    const { path: sitePath, environment, cdk } = this.props;
+    const baseServerConfig = {
+      description: "Next.js Server",
       environment: {
         CACHE_BUCKET_NAME: bucket.bucketName,
         CACHE_BUCKET_KEY_PREFIX: "_cache",
         CACHE_BUCKET_REGION: Stack.of(this).region,
       },
-      layers: this.isPerRouteLoggingEnabled()
-        ? [
-            LayerVersion.fromLayerVersionArn(
-              this,
-              "SSTExtension",
-              cdk?.server?.architecture?.name === Architecture.X86_64.name
-                ? `arn:aws:lambda:${stack.region}:226609089145:layer:sst-extension-amd64:${LAYER_VERSION}`
-                : `arn:aws:lambda:${stack.region}:226609089145:layer:sst-extension-arm64:${LAYER_VERSION}`
-            ),
-          ]
-        : undefined,
-    };
-    this.removeSourcemaps();
+    }
+    //@ts-expect-error
+    const functionCdkOverrides = (cdk?.servers?.[key] ?? {}) as InterpolatedCdkProp<{}>;
+    return {
+      type: "function" as const,
+      constructId: `${key}ServerFunction`,
+      function: {
+        ...baseServerConfig,
+        handler: fn.handler,
+        bundle: path.join(sitePath, fn.bundle),
+        runtime: "nodejs18.x" as const,
+        architecture: Architecture.ARM_64,
+        memorySize: 1024,
```

Comment:
We should default the lambdas to 1536MB, see: https://docs.aws.amazon.com/lambda/latest/operatorguide/computing-power.html

tl;dr - 1536 MB is almost as cheap as 128 MB but you also get max speed.

Reply:
I don't believe that's strictly true. The AWS table provided is an example, they explain that by increasing memory you also increase vCPU given and that for CPU bound operations it will be cheaper as the function will run faster. 

But **if** it doesn't complete faster, it'll just be more expensive. That's why they've got a complicated tool to measure and profile your services ideal memory:
- https://docs.aws.amazon.com/lambda/latest/operatorguide/profile-functions.html

Reply:
That's fair. It's a bit tricky b/c there's been several people getting low latency from certain dependencies like prisma and increasing the memory helped minimize it. But everyone's mileage may vary.